packages/wonder-blocks-dropdown/components/multi-select-menu.js

```diff
@@ -0,0 +1,227 @@
+     * selected, the SelectBox would display "2 students"
+     */
+    selectItemType: string,
+
+    /**
+     * Whether to display shortcuts for Select All and Select None.
+     */
+    displaySelectionShortcuts?: boolean,
+
+    /**
+     * Optional placeholder for the opening component when there are no items
+     * selected.
+     */
+    placeholder?: string,
+
+    /**
+     * Whether this menu should be left-aligned or right-aligned with the
+     * opener component. Defaults to left-aligned.
+     */
+    alignment?: "left" | "right",
+
+    /**
+     * Whether this menu is disabled. A disabled menu may not be opened and
+     * does not support interaction. Defaults to false.
+     */
+    disabled?: boolean,
+
+    /**
+     * Whether to display the "light" version of this component instead, for
+     * use when the item is used on a dark background.
+     */
+    light?: boolean,
+
+    /**
+     * Optional styling to add.
+     */
+    style?: any,
+};
+
+type State = {
+    /**
+     * Whether or not menu is open.
+     */
+    open: boolean,
+};
+
+export default class MultiSelectMenu extends React.Component<Props, State> {
+    static defaultProps = {
+        alignment: "left",
+        disabled: false,
+        displaySelectionShortcuts: false,
+        light: false,
+    };
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            open: false,
+        };
+    }
+
+    toggleMenu() {
+        this.setState((prevState) => ({
+            open: !prevState.open,
+        }));
+    }
+
+    handleSelected(selectedValue: string, selectionState: boolean) {
+        const {onChange} = this.props;
+
+        onChange([{value: selectedValue, selected: selectionState}]);
+    }
+
+    handleSelectAll() {
+        const {items, onChange, selectedItems} = this.props;
+        const update = items
+            .filter((item) => !selectedItems.includes(item.value))
+            .map((item) => {
+                return {
+                    value: item.value,
+                    selected: true,
+                };
+            });
+        onChange(update);
+    }
+
+    handleSelectNone() {
+        const {items, onChange, selectedItems} = this.props;
+        const update = items
+            .filter((item) => selectedItems.includes(item.value))
+            .map((item) => {
+                return {
+                    value: item.value,
+                    selected: false,
+                };
+            });
+        onChange(update);
+    }
+
+    render() {
+        const {
+            alignment,
+            displaySelectionShortcuts,
+            disabled,
+            items,
+            light,
+            placeholder,
+            selectItemType,
+            selectedItems,
+            style,
+        } = this.props;
+
+        const {open} = this.state;
+
+        // TODO(sophie): figure out how to configure plurals for i18n
```

Comment:
We'll need a copy of the `i18n.ngettext()` function from `i18n.js` in `webapp`.  Probably best to add it to `wonder-blocks-core`... or maybe we can create an `i18n` package.  This can be a separate PR.